src/main/plugin/iso19139.ca.HNAP/convert/thesaurus-transformation.xsl

```diff
@@ -179,12 +240,36 @@
                                         else if ($listOfLanguage)
                                         then $listOfLanguage
                                         else ''" />
-
       <xsl:copy-of
         select="geonet:add-thesaurus-info-2($currentThesaurus, $withThesaurusAnchor, /root/gui/thesaurus/thesauri, not(/root/request/keywordOnly), $lang)"/>
     </gmd:MD_Keywords>
   </xsl:template>
 
+  <!-- given a thesarus and language, find the most appropriate title-->
+  <xsl:function name="geonet:getThesaurusTitle">
+    <xsl:param name="thesarus" />
+    <xsl:param name="lang1" />
+
+    <xsl:variable name="lang" select="lower-case($lang1)" />
+    <xsl:variable name="lang_2letter" select="lower-case(XslUtilHnap:twoCharLangCode($lang))" />
```

Comment:
@davidblasby 
This line causes the following error in the build when executing the unit tests

Error at xsl:variable on line 254 column 97 of thesaurus-transformation.xsl:
  XPST0017: XPath syntax error at char 45 on line 254 in {...ilHnap:twoCharLangCode($lan...}:
    Cannot find a matching 1-argument function named
  {java:ca.gc.schema.iso19139hnap.util.XslUtilHnap}twoCharLangCode()

It fails in the GetKeywordByIdTest

Could you please correct this.

Thank you

Reply:
This function exists;
https://github.com/metadata101/iso19139.ca.HNAP/blob/3.10.x/src/main/java/ca/gc/schema/iso19139hnap/util/XslUtilHnap.java#L172

I'm going to guess you are running the GN unit tests when you have HNAP installed.  
You cannot do that - the modules do NOT have dependencies on HNAP (specifically, the jar), but pick up hnap because the schema files (but not the jar) been installed in the web app.  This causes errors.

Reply:
@davidblasby 
Documentation specifically has instruction where it is acceptable to include the module and do a build (with or without tests) - it is even with the HNAP documentation on the main.[readme.md]https://github.com/metadata101/iso19139.ca.HNAP)

In our case we have added some extra java code that we compile as part of the regular build and this code has unit test which are also executed as part of our build pipeline.  So in our case we need to be able to build and run unit test for the whole system or we cannot be deploy changes.  This is standard practice.

So if the build with these changes generate errors then maybe the unit tests are flawed.  There seems to be a bug  where the unit test are being executed on all the schema plugins instead of only the ones that are in the geonetwork core code.  If it is going to be executed on other plugins then the unit test should ensure that they have all the dependencies fully loaded. Is it possible to fix the unit test?

Reply:
@ianwallen - I haven't been able to run the GN unit tests with HNAP installed for a very long time (I dont think ever).  This is an architectual issue in GN with how it manages schemas and dependencies - its a bit of a mess as you're seen.

However, you should be able to resolve it by adding HNAP as a dependency (test scope) to the project that has failing test cases.

Reply:
@davidblasby good idea about adding the dependency.  I have created a new PR that will do that. see https://github.com/geonetwork/core-geonetwork/pull/4835

Reply:
@davidblasby looks like there was another bug with the spring files once I added the dependency.  I created pull request https://github.com/metadata101/iso19139.ca.HNAP/pull/61 to correct this issue.